picard/ui/options/maintenance.py

```diff
@@ -0,0 +1,168 @@
+    register_options_page,
+)
+from picard.ui.ui_options_maintenance import Ui_MaintenanceOptionsPage
+
+
+class MaintenanceOptionsPage(OptionsPage):
+
+    NAME = "maintenance"
+    TITLE = N_("Maintenance")
+    PARENT = None
+    SORT_ORDER = 99
+    ACTIVE = True
+    HELP_URL = '/config/options_maintenance.html'
+
+    options = []
+
+    INSTRUCTIONS = N_(
+        "This allows you to remove unused option settings from the configuration.\n\n"
+        "Settings that are found in the configuration that do not appear on any option "
+        "settings page will be listed below. If your configuration does not contain any "
+        "unused option settings, then the list will be empty.\n\n"
+        "Note that unused option settings could come from plugins that have been removed, "
+        "so please be careful to not remove settings that you may want to use later when "
+        "the plugin is reloaded.  Options belonging to plugins that are loaded but "
+        "currently disabled will not be listed for possible removal.\n\n"
+        "To remove one or more settings, select the settings to remove by checking the "
+        "box next to the settings, and then enable the removal by checking the \"Remove "
+        "selected options\" box. When you choose \"Make It So!\" to save your option "
+        "settings, the selected items will be removed."
+    )
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.ui = Ui_MaintenanceOptionsPage()
+        self.ui.setupUi(self)
+        self.ui.description.setText(_(self.INSTRUCTIONS))
+        self.ui.tableWidget.setHorizontalHeaderLabels([_("Option"), _("Value")])
+
+    def load(self):
+        from picard.plugin import _extension_points
+        config = get_config()
+
+        # Setting options from all option pages and loaded plugins (including plugins currently disabled).
+        key_options = set(config.setting.as_dict().keys())
+
+        # All setting options included in the INI file.
+        file_options = set([key[8:] for key in config.allKeys() if key[:8] == 'setting/'])
+
+        # Combine all page and plugin settings with required options not appearing in option pages.
+        current_options = set([
+            # Include options that are required but are not entered directly from the options pages.
+            'file_renaming_scripts',
+            'selected_file_naming_script_id',
+            'log_verbosity',
+            # Items missed if TagsCompatibilityWaveOptionsPage does not register.
+            'remove_wave_riff_info',
+            'wave_riff_info_encoding',
+            'write_wave_riff_info',
+        ]).union(key_options)
+
+        def _add_options_from_page(page, current_options):
+            if not hasattr(page, 'options'):
+                return
+            for option in page.options:
+                if option.section == "setting":
+                    current_options.add(option.name)
+
+        for ep in _extension_points:
```

Comment:
Disabled plugins will not be present in this list (see https://github.com/metabrainz/picard/blob/f950188f1f460b4e0b7a5cf599868e9b1c60c43b/picard/plugin.py#L102-L107).
Is this intended?

Reply:
> Disabled plugins will not be present in this list

That's what I though too, but in my testing (Windows 7 running from source code) the disabled plugin settings still appeared in the list.  Only the plugins that hadn't been downloaded (installed) didn't appear.

Reply:
I just tested, with ReplayGain plugin, it registers following options:

```
replaygain_vorbisgain_command
replaygain_vorbisgain_options
replaygain_mp3gain_command
replaygain_mp3gain_options
replaygain_metaflac_command
replaygain_metaflac_options
replaygain_wvgain_command
replaygain_wvgain_options
```

If I print `option.name` (line 100), they do show when plugin is enabled, but they don't if not. That's what it is expected from the code above. So `current_options` does actually differs whether the plugin is enabled or disabled.

That said, it doesn't change the result as none of those options are in final list:

```python
orphan_options = file_options.difference(current_options)
```

When plugin is enabled, `current_options` set is bigger, but `file_options` is unchanged:

```python
>>> file_options = {1, 2, 3}
>>> current_options = {2, 3, 4}
>>> file_options.difference(current_options)
{1}
>>> current_options = {2, 3, 4, 5}
>>> file_options.difference(current_options)
{1}
```

Reply:
Okay, now I think I understand.  I was only looking at the `orphan_options` difference set and it wasn't changing.  I didn't catch the change in the numbers of the current options.

Is this okay as-is, or would you like me to change it so that the `current_options` number also includes the disabled plugins?